ccpp/data/GFS_typedefs.F90

```diff
@@ -5394,29 +5420,64 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
 
     ENDIF !}
 
-    ! To ensure that these values match what's in the physics,
-    ! array sizes are compared during model init in GFS_phys_time_vary_init()
-    !
-    ! from module ozinterp
-    if (Model%ntoz>0) then
-       if (Model%oz_phys) then
-          levozp   = 80
-          oz_coeff = 4
-       else if (Model%oz_phys_2015) then
-          levozp   = 53
-          oz_coeff = 6
-       else
-          write(*,*) 'Logic error, ntoz>0 but no ozone physics selected'
-          stop
-       end if
+    !--- NRL ozone physics
+    if (Model%ntoz > 0) then
+       ! Get dimensions from data file
+       open(unit=Model%kozpl,file='global_o3prdlos.f77', form='unformatted', convert='big_endian')
```

Comment:
I don't understand how this is working. `Model%kozpl` shouldn't be defined at this point. You've set the local `kozpl`, but haven't initialized the `Model` component yet, right? Same for `kozc`.

Reply:
Great point. My guess is that it works with whatever garbage it's initialized to. I will fix this. 
Another thing, I don't think these file identifiers need to be interstitials anymore, as they aren't used in the physics, just in this module. I will also remove.

Reply:
outdated